src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2537,13 +2536,13 @@ CoPartitionedTables(Oid firstRelationId, Oid secondRelationId)
 	FmgrInfo *comparisonFunction = firstTableCache->shardIntervalCompareFunction;
 
 	/* reference tables are always & only copartitioned with reference tables */
-	if (firstTableCache->partitionMethod == DISTRIBUTE_BY_NONE &&
-		secondTableCache->partitionMethod == DISTRIBUTE_BY_NONE)
+	if (IsCitusTableTypeCacheEntry(firstTableCache, REFERENCE_TABLE) &&
```

Comment:
to be compatible with other planner functions, let's use NONE_DIST_KEY in this function?